sdk/communication/Azure.Communication.CallingServer/src/ConversationClient.cs

```diff
@@ -0,0 +1,639 @@
+                Argument.CheckNotNull(tokenCredential, nameof(tokenCredential)),
+                options ?? new CallClientOptions())
+        { }
+
+        #endregion
+
+        #region private constructors
+
+        private ConversationClient(ConnectionString connectionString, CallClientOptions options)
+            : this(connectionString.GetRequired("endpoint"), options.BuildHttpPipeline(connectionString), options)
+        { }
+
+        private ConversationClient(string endpoint, TokenCredential tokenCredential, CallClientOptions options)
+            : this(endpoint, options.BuildHttpPipeline(tokenCredential), options)
+        { }
+
+        private ConversationClient(string endpoint, AzureKeyCredential keyCredential, CallClientOptions options)
+            : this(endpoint, options.BuildHttpPipeline(keyCredential), options)
+        { }
+
+        private ConversationClient(string endpoint, HttpPipeline httpPipeline, CallClientOptions options)
+        {
+            _clientDiagnostics = new ClientDiagnostics(options);
+            RestClient = new ConversationRestClient(_clientDiagnostics, httpPipeline, endpoint, options.ApiVersion);
+        }
+
+        #endregion
+
+        /// <summary>Initializes a new instance of <see cref="ConversationClient"/> for mocking.</summary>
+        protected ConversationClient()
+        {
+            _clientDiagnostics = null;
+            RestClient = null;
+        }
+
+        /// Join the call using conversation id.
+        /// <param name="conversationId"> The conversation id that can be a group id or a encoded conversation url retrieve from client. </param>
+        /// <param name="source"> The source identity. </param>
+        /// <param name="callOptions"> The call Options. </param>
+        /// <param name="cancellationToken"> The cancellation token. </param>
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="conversationId"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="callOptions"/> is null.</exception>
+        public virtual async Task<Response<JoinCallResponse>> JoinCallAsync(string conversationId, CommunicationIdentifier source, JoinCallOptions callOptions, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallClient)}.{nameof(JoinCall)}");
+            scope.Start();
+            try
+            {
+                Argument.AssertNotNull(callOptions, nameof(callOptions));
+
+                return await RestClient.JoinCallAsync(
+                    conversationId: conversationId,
+                    source: CommunicationIdentifierSerializer.Serialize(source),
```

Comment:
Do you also need to check that the source is not null since you are serializing it?

Reply:
good catch, this will get addressed in the update.